src/include/server/mir/input/input_event_transformer.h

```diff
@@ -53,12 +75,15 @@ class InputEventTransformer
         EventBuilder* builder,
         EventDispatcher const& dispatcher);
 
-    void append(std::weak_ptr<Transformer> const&);
-    bool remove(std::shared_ptr<Transformer> const&);
+    /// Appends the transformer to be used in `transform`
+    /// Precondition: Transformer isn't already registered.
```

Comment:
Do we need this precondition?

1. We don't check the precondition in the implementation;
2. There's no way for the user of the API to verify the precondition; and,
3. There's no obvious difficulty with registering a transformer twice (see below)

I realize that (pseudo code):

```c++
auto ra = append(a);
auto rb = append(b);
{
    auto rc = append(a);
}
```

Will leave the transformers as `{b, a}`, not the slightly more intuitive `{a, b}`, but how much of a problem is that?